api/document/js/list-utils.ts

```diff
@@ -1,29 +1,83 @@
 import { Node, ResolvedPos } from 'prosemirror-model';
+import { Transaction } from 'prosemirror-state';
+import * as repeatString from 'repeat-string';
+import * as romanize from 'romanize';
 
-import schema from './schema';
+import schema, { factory } from './schema';
+import { walkUpUntil } from './util';
 
 const defaultBullet = '●';
 const bulletFollows = {
   '●': '○',
   '○': '■',
 };
 
-function parentListItem(pos: ResolvedPos): Node | void {
-  let depth = pos.depth;
-  // Walk up the document until we hit a list
-  while (depth >= 0) {
-    const node = pos.node(depth);
-    if (node.type === schema.nodes.listitem) {
-      return node;
-    }
-    depth -= 1;
-  }
-}
-
 export function deeperBullet(pos: ResolvedPos): string {
-  const parListItem = parentListItem(pos);
-  if (parListItem) {
+  const liDepth = walkUpUntil(pos, node => node.type === schema.nodes.listitem);
+  if (liDepth >= 0) {
+    const parListItem = pos.node(liDepth);
     return bulletFollows[parListItem.attrs.marker] || defaultBullet;
   }
   return defaultBullet;
 }
+
+function makeIntToLetter(initial: string) {
+  const offset = initial.charCodeAt(0);
+
+  return (idx: number) => {
+    const count = Math.floor(idx / 26) + 1;
+    const letter = String.fromCharCode(offset + idx % 26);
```

Comment:
I think the meaning of `26` is fairly obvious from context here, though ordinarily I'd suggest making it a named constant like `NUM_LETTERS_IN_ALPHABET`.  Not sure if this would actually be helpful in this case or not.

Reply:
I'll play with it a bit to see if there's something helpful and concise. I'll also add a test around the boundary.